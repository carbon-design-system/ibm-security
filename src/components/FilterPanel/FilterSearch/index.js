/**
 * @file Filter search entry point.
 * @copyright IBM Security 2020
 */

import FilterSearch from './FilterSearch';

export default FilterSearch;
