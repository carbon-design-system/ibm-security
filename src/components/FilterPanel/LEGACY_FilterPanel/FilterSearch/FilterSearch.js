/**
 * @file Filter search.
 * @copyright IBM Security 2019
 */

import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { Component } from 'react';

import { getComponentNamespace } from '../../../../globals/namespace';
import * as defaultLabels from '../../../../globals/nls';

import FilterSelector from '../FilterSelector';
import { filterDataPropTypes, getFiltersArray } from '../FilterPanelUtilities';
import Search from '../../../Search';

export const namespace = getComponentNamespace('filter-search');

class FilterSearch extends Component {
  state = {
    searchValue: this.props.initialSearchValue,
  };

  /**
   * Handles filter toggling.
   * @param {Filter} filter Filter object to toggle.
   */
  onFilterSelection = filter => {
    this.setState({ searchValue: '' });
    this.props.onChange(filter);
  };

  /**
   * Handles search value change.
   * @param {KeyboardEvent} event Event generated by input change.
   */
  onFilterSearch = event => {
    this.setState({ searchValue: event.target.value });
  };

  /**
   * Filters filter labels based on a given search value.
   * @param {string} string String to match filters labels against.
   */
  searchFilterData = string => {
    if (!string || string.length === 0) {
      return [];
    }

    const { filterData } = this.props;
    const filters = getFiltersArray(filterData);
    const stringToMatch = string.trim().toLowerCase();

    return filters
      .filter(filter => filter.name.toLowerCase().includes(stringToMatch))
      .reduce((subcategories, filter) => {
        const filterParentId = filter.parent;
        const subcategoryFilters = subcategories[filterParentId]
          ? [...subcategories[filterParentId]]
          : [];

        return {
          ...subcategories,
          [filterParentId]: [...subcategoryFilters, filter.id],
        };
      }, {});
  };

  render() {
    const {
      filterData,
      labels,
      filterSearchLabel,
      noFiltersResultsLabel,
    } = this.props;
    const { searchValue } = this.state;
    const resultsPanelVisible = searchValue && searchValue.length > 0;
    const searchResults = this.searchFilterData(searchValue);

    const resultsContainerClassNames = classnames(`${namespace}__results`, {
      [`${namespace}__results--visible`]: resultsPanelVisible,
    });

    const componentLabels = {
      ...defaultLabels.labels,
      ...defaultLabels.filterFalsey({
        FILTER_PANEL_SEARCH_LABEL: filterSearchLabel,
        FILTER_PANEL_SEARCH_NO_RESULTS_LABEL: noFiltersResultsLabel,
      }),
      ...labels,
    };

    const {
      FILTER_PANEL_SEARCH_NO_RESULTS_LABEL,
      SEARCH_CLOSE_BUTTON,
      SEARCH_LABEL,
      SEARCH_PLACEHOLDER_LABEL,
    } = {
      ...componentLabels,
      ...defaultLabels.filterFalsey({
        SEARCH_LABEL: componentLabels.FILTER_PANEL_SEARCH_LABEL,
        SEARCH_PLACEHOLDER_LABEL: componentLabels.FILTER_PANEL_SEARCH_LABEL,
        SEARCH_CLOSE_BUTTON: componentLabels.FILTER_PANEL_SEARCH_CLOSE_BUTTON,
      }),
    };

    return (
      <div className={namespace}>
        <Search
          className={`${namespace}__input`}
          closeButtonLabelText={SEARCH_CLOSE_BUTTON}
          labelText={SEARCH_LABEL}
          onChange={this.onFilterSearch}
          placeHolderText={SEARCH_PLACEHOLDER_LABEL}
          value={searchValue}
          size="sm"
        />
        {resultsPanelVisible && (
          <div className={resultsContainerClassNames}>
            {Object.keys(searchResults).length === 0 ? (
              <span className={`${namespace}__results--none`}>
                {FILTER_PANEL_SEARCH_NO_RESULTS_LABEL}
              </span>
            ) : (
              <ul className={`${namespace}__list`}>
                {Object.keys(searchResults)
                  .map(subcategoryId => filterData.subcategories[subcategoryId])
                  .map(subcategory => (
                    <li
                      key={`search-subcategory-${subcategory.id}`}
                      className={`${namespace}__subcategory`}
                    >
                      <h3 className={`${namespace}__subcategory__title`}>
                        {subcategory.name}
                      </h3>
                      <ul className={`${namespace}__list`}>
                        {searchResults[subcategory.id].map(filterId => {
                          const filter = filterData.filters[filterId];
                          return (
                            <li key={`search-filter-${filter.id}`}>
                              <FilterSelector
                                filter={filter}
                                onChange={this.onFilterSelection}
                              />
                            </li>
                          );
                        })}
                      </ul>
                    </li>
                  ))}
              </ul>
            )}
          </div>
        )}
      </div>
    );
  }
}

FilterSearch.propTypes = {
  /** @type {string} Initial search value */
  initialSearchValue: PropTypes.string,

  /** @type {(filter: Filter) => {}} Callback function when any filter is toggled */
  onChange: PropTypes.func,

  /** @type {FilterData}  Filter data to used to render panel */
  filterData: filterDataPropTypes.isRequired,

  /** @type {string} Label for filter search input */
  filterSearchLabel: PropTypes.string,

  /** @type {string} Label for filter search when no filters are found */
  noFiltersResultsLabel: PropTypes.string,

  /** @type {object} Externalized global labels */
  labels: defaultLabels.propType,
};

FilterSearch.defaultProps = {
  onChange: () => {},
  initialSearchValue: '',
  filterSearchLabel: '',
  noFiltersResultsLabel: '',
  labels: {},
};

export default FilterSearch;
