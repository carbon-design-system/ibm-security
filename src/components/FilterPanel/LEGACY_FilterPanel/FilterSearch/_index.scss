////
/// Legacy filter search styles.
/// @group filter-panel
/// @copyright IBM Security 2019
////

@import '../../../Component/mixins';

@import 'mixins';

@include namespace($selector: filter-search) {
  @include LEGACY-filter-search;
}
