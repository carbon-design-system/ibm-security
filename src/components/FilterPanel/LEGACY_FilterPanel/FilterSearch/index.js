/**
 * @file Filter Search entry point.
 * @copyright IBM Security 2019
 */

import FilterSearch from './FilterSearch';

export default FilterSearch;
