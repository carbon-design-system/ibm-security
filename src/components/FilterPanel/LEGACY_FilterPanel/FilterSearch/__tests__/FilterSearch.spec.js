/**
 * @file Filter Search tests.
 * @copyright IBM Security 2019
 */

import React from 'react';
import { shallow } from 'enzyme';
import FilterSearch from '../FilterSearch';
import {
  filterData,
  noFiltersResultsLabel,
  filterSearchLabel,
  labels,
} from '../../../_mocks_';

const props = {
  filterData,
  labels,
  onChange: () => {},
};

const uniqueProps = {
  noFiltersResultsLabel,
  filterSearchLabel,
};

describe('FilterSearch', () => {
  describe('Rendering', () => {
    it('renders', () =>
      expect(shallow(<FilterSearch {...props} />)).toMatchSnapshot());

    it('accepts initial search value', () =>
      expect(
        shallow(<FilterSearch {...props} initialSearchValue="filter 10" />)
      ).toMatchSnapshot());

    it('displays message when no filters match search value', () => {
      const wrapper = shallow(
        <FilterSearch
          {...props}
          initialSearchValue="should not match a filter"
        />
      );
      expect(wrapper).toMatchSnapshot();
    });

    it('partially matches any string', () => {
      const wrapper = shallow(
        <FilterSearch {...props} initialSearchValue="FiLTer 1" />
      );
      expect(wrapper).toMatchSnapshot();
      expect(wrapper).toMatchSnapshot();
    });

    it('should not overwrite `labels` when unique props defined', () => {
      const wrapper = shallow(<FilterSearch {...props} {...uniqueProps} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('displays search results when value typed in', () => {
      const wrapper = shallow(<FilterSearch {...props} />);
      wrapper
        .find('Search')
        .simulate('change', { target: { value: 'filter 10' } });
      expect(wrapper).toMatchSnapshot();
    });

    it('hides results when filter has been toggled', () => {
      const wrapper = shallow(
        <FilterSearch {...props} initialSearchValue="filter 10" />
      );
      wrapper
        .find('FilterSelector')
        .at(0)
        .simulate('change');
      expect(wrapper).toMatchSnapshot();
    });

    it('invokes onChange when filter is toggled', () => {
      const onChange = jest.fn();
      const wrapper = shallow(
        <FilterSearch
          {...props}
          onChange={onChange}
          initialSearchValue="filter 10"
        />
      );
      const filterSelector = wrapper.find('FilterSelector').at(0);
      const filterObject = filterSelector.prop('filter');

      filterSelector.simulate('change', filterObject);
      expect(onChange).toHaveBeenCalledWith(filterObject);
    });
  });
});
