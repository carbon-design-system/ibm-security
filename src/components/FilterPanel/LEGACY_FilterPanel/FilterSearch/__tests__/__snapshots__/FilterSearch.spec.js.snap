// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSearch Events displays search results when value typed in 1`] = `
<div
  className="security--filter-search"
>
  <Search
    className="security--filter-search__input"
    closeButtonLabelText="Clear search"
    labelText="Find filter..."
    onChange={[Function]}
    placeHolderText="Find filter..."
    small={true}
    type="text"
    value="filter 10"
  />
  <div
    className="security--filter-search__results security--filter-search__results--visible"
  >
    <ul
      className="security--filter-search__list"
    >
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-TRUNCATED"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Truncated subcategory
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-TRUNCATED:FILTER_10"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_10",
                  "name": "Filter 10",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`FilterSearch Events hides results when filter has been toggled 1`] = `
<div
  className="security--filter-search"
>
  <Search
    className="security--filter-search__input"
    closeButtonLabelText="Clear search"
    labelText="Find filter..."
    onChange={[Function]}
    placeHolderText="Find filter..."
    small={true}
    type="text"
    value=""
  />
</div>
`;

exports[`FilterSearch Rendering accepts initial search value 1`] = `
<div
  className="security--filter-search"
>
  <Search
    className="security--filter-search__input"
    closeButtonLabelText="Clear search"
    labelText="Find filter..."
    onChange={[Function]}
    placeHolderText="Find filter..."
    small={true}
    type="text"
    value="filter 10"
  />
  <div
    className="security--filter-search__results security--filter-search__results--visible"
  >
    <ul
      className="security--filter-search__list"
    >
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-TRUNCATED"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Truncated subcategory
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-TRUNCATED:FILTER_10"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_10",
                  "name": "Filter 10",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`FilterSearch Rendering displays message when no filters match search value 1`] = `
<div
  className="security--filter-search"
>
  <Search
    className="security--filter-search__input"
    closeButtonLabelText="Clear search"
    labelText="Find filter..."
    onChange={[Function]}
    placeHolderText="Find filter..."
    small={true}
    type="text"
    value="should not match a filter"
  />
  <div
    className="security--filter-search__results security--filter-search__results--visible"
  >
    <span
      className="security--filter-search__results--none"
    >
      No filter matches your search
    </span>
  </div>
</div>
`;

exports[`FilterSearch Rendering partially matches any string 1`] = `
<div
  className="security--filter-search"
>
  <Search
    className="security--filter-search__input"
    closeButtonLabelText="Clear search"
    labelText="Find filter..."
    onChange={[Function]}
    placeHolderText="Find filter..."
    small={true}
    type="text"
    value="FiLTer 1"
  />
  <div
    className="security--filter-search__results security--filter-search__results--visible"
  >
    <ul
      className="security--filter-search__list"
    >
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-TRUNCATED"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Truncated subcategory
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-TRUNCATED:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_1",
                  "name": "Filter 1",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_10"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_10",
                  "name": "Filter 10",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_11"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_11",
                  "name": "Filter 11",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_12"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_12",
                  "name": "Filter 12",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_13"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_13",
                  "name": "Filter 13",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_14"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_14",
                  "name": "Filter 14",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_15"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_15",
                  "name": "Filter 15",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_16"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_16",
                  "name": "Filter 16",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_17"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_17",
                  "name": "Filter 17",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_18"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_18",
                  "name": "Filter 18",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_19"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_19",
                  "name": "Filter 19",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-ZERO_FILTERS_COUNT"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory with filters with no count
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-ZERO_FILTERS_COUNT:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 0,
                  "enabled": false,
                  "id": "ZERO_FILTERS_COUNT:FILTER_1",
                  "name": "Filter 1",
                  "parent": "ZERO_FILTERS_COUNT",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-OPEN"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Open subcategory
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-OPEN:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "OPEN:FILTER_1",
                  "name": "Filter 1",
                  "parent": "OPEN",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_1"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 1
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_1:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 5,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_1:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_1",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_2"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 2
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_2:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 15,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_2:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_2",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_3"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 3
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_3:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 13,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_3:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_3",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_4"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 4
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_4:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 26,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_4:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_4",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_5"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 5
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_5:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_5:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_5",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_6"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 6
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_6:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 24,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_6:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_6",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`FilterSearch Rendering partially matches any string 2`] = `
<div
  className="security--filter-search"
>
  <Search
    className="security--filter-search__input"
    closeButtonLabelText="Clear search"
    labelText="Find filter..."
    onChange={[Function]}
    placeHolderText="Find filter..."
    small={true}
    type="text"
    value="FiLTer 1"
  />
  <div
    className="security--filter-search__results security--filter-search__results--visible"
  >
    <ul
      className="security--filter-search__list"
    >
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-TRUNCATED"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Truncated subcategory
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-TRUNCATED:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_1",
                  "name": "Filter 1",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_10"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_10",
                  "name": "Filter 10",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_11"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_11",
                  "name": "Filter 11",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_12"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_12",
                  "name": "Filter 12",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_13"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_13",
                  "name": "Filter 13",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_14"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_14",
                  "name": "Filter 14",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_15"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_15",
                  "name": "Filter 15",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_16"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_16",
                  "name": "Filter 16",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_17"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_17",
                  "name": "Filter 17",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_18"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_18",
                  "name": "Filter 18",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
          <li
            key="search-filter-TRUNCATED:FILTER_19"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "TRUNCATED:FILTER_19",
                  "name": "Filter 19",
                  "parent": "TRUNCATED",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-ZERO_FILTERS_COUNT"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory with filters with no count
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-ZERO_FILTERS_COUNT:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 0,
                  "enabled": false,
                  "id": "ZERO_FILTERS_COUNT:FILTER_1",
                  "name": "Filter 1",
                  "parent": "ZERO_FILTERS_COUNT",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-OPEN"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Open subcategory
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-OPEN:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "OPEN:FILTER_1",
                  "name": "Filter 1",
                  "parent": "OPEN",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_1"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 1
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_1:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 5,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_1:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_1",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_2"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 2
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_2:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 15,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_2:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_2",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_3"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 3
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_3:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 13,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_3:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_3",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_4"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 4
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_4:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 26,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_4:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_4",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_5"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 5
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_5:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 10,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_5:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_5",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
      <li
        className="security--filter-search__subcategory"
        key="search-subcategory-CATEGORY_2:SUBCATEGORY_6"
      >
        <h3
          className="security--filter-search__subcategory__title"
        >
          Subcategory 6
        </h3>
        <ul
          className="security--filter-search__list"
        >
          <li
            key="search-filter-CATEGORY_2:SUBCATEGORY_6:FILTER_1"
          >
            <FilterSelector
              filter={
                Object {
                  "count": 24,
                  "enabled": false,
                  "id": "CATEGORY_2:SUBCATEGORY_6:FILTER_1",
                  "name": "Filter 1",
                  "parent": "CATEGORY_2:SUBCATEGORY_6",
                }
              }
              onChange={[Function]}
            />
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`FilterSearch Rendering renders 1`] = `
<div
  className="security--filter-search"
>
  <Search
    className="security--filter-search__input"
    closeButtonLabelText="Clear search"
    labelText="Find filter..."
    onChange={[Function]}
    placeHolderText="Find filter..."
    small={true}
    type="text"
    value=""
  />
</div>
`;

exports[`FilterSearch Rendering should not overwrite \`labels\` when unique props defined 1`] = `
<div
  className="security--filter-search"
>
  <Search
    className="security--filter-search__input"
    closeButtonLabelText="Clear search"
    labelText="Find filter..."
    onChange={[Function]}
    placeHolderText="Find filter..."
    small={true}
    type="text"
    value=""
  />
</div>
`;
